client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,286 @@
+import { Button } from '@wordpress/components';
+import { filterSortAndPaginate, Operator } from '@wordpress/dataviews';
+import { Icon, link, linkOff, plugins, trash } from '@wordpress/icons';
+import { useTranslate } from 'i18n-calypso';
+import { find } from 'lodash';
+import { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import { navigate } from 'calypso/lib/navigate';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { PluginActions } from '../hooks/types';
+import PluginActionStatus from '../plugin-management-v2/plugin-action-status';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	pluginsWithUpdates: Array< Plugin >;
+	activePlugins: Array< Plugin >;
+	inactivePlugins: Array< Plugin >;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export const PLUGINS_STATUS = {
+	ACTIVE: 1,
+	INACTIVE: 2,
+	UPDATE: 3,
+};
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	pluginsWithUpdates,
+	activePlugins,
+	inactivePlugins,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( ( state ) => {
+		return getPluginActionStatuses( state );
+	} );
+
+	// Add flags for plugins status: active, inactive, updates
+	currentPlugins.map( ( plugin ) => {
+		plugin.status = [];
+
+		if ( find( pluginsWithUpdates, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.UPDATE );
+		}
+
+		if ( find( inactivePlugins, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.INACTIVE );
+		}
+
+		if ( find( activePlugins, { slug: plugin.slug } ) ) {
+			plugin.status.push( PLUGINS_STATUS.ACTIVE );
+		}
+
+		return plugin;
+	} );
+
+	const translate = useTranslate();
+
+	const fields = useMemo(
```

Comment:
We could move the fields and actions to separate hooks. That usually helps with understanding what level of optimization / memoization is necessary and what not.

Reply:
Interesting, [I extracted them here](https://github.com/Automattic/wp-calypso/pull/95561/commits/582540f4231916156627bdcf3a4fea05951159b3). Can you confirm if is that what you meant @tyxla?

If so, since we're not sharing these hooks with other components, I'm curious to understand what's the advantage of this strategy besides making the `plugins-list-dataviews` easer to read?

Reply:
There's no immediate benefit other than separation of concerns, but it can help diagnose if there are frequent (potentially expensive) re-renders happening because of what's going on in these hooks.